src/transformers/trainer.py

```diff
@@ -1619,9 +1622,14 @@ def optimizer_hook(param):
                     "You need to have `torch>2.4` in order to use torch 4-bit optimizers. "
                     "Install it with `pip install --upgrade torch` it is available on pipy. Otherwise, you need to install torch nightly."
                 )
-            from torchao.prototype.low_bit_optim import AdamW4bit
+            from torchao.prototype.low_bit_optim import AdamW4bit, AdamW8bit
 
-            optimizer_cls = AdamW4bit
+            if args.optim == OptimizerNames.ADAMW_TORCH_4BIT:
+                optimizer_cls = AdamW4bit
+            elif args.optim == OptimizerNames.ADAMW_TORCH_8BIT:
+                optimizer_cls = AdamW8bit
+            else:
+                raise ValueError("Invalid optimizer")
```

Comment:
We don't need this as we already checked for args.optim previously

Reply:
Hi, I am mimicking 
![image](https://github.com/user-attachments/assets/4a77aed9-73db-4494-a2f5-3a41b399e654)

Reply:
Sure, we can keep it like this. It was only a small nit.